pkg/build/builder/common.go

```diff
@@ -81,3 +84,47 @@ func updateBuildRevision(c client.BuildInterface, build *api.Build, sourceInfo *
 		glog.Warningf("An error occurred saving build revision: %v", err)
 	}
 }
+
+// runScriptInContainer runs a Bash script in an ephemeral Docker container
+// started off imageID, and returns an error if the script cannot be run or
+// returns a non-zero exit code.
+func runScriptInContainer(client DockerClient, script, imageID string) error {
+	glog.Infof("Running post commit hook with image %s ...", imageID)
+
+	// Read CPU and memory limits from /sys/fs/cgroup.
+	// These limits are used to enforce limits on the container that runs
+	// the post commit hook.
+	r := intFromFileReader{}
+	cpuShares := r.read("/sys/fs/cgroup/cpu,cpuacct/cpu.shares")
```

Comment:
this is not necessarily going into s2i.  I was intending to do the filesystem reading in the openshift code and pass the cgroup limit values to s2i which would just use them when launching the assemble container.  the s2i code would not be responsible for reading the cgroup limit files.